packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -4540,6 +4542,47 @@ QUnit.test('form should be dirty when some editors are dirty', function(assert)
     assert.strictEqual(form.option('isDirty'), false, 'form is not dirty when all editors are back to pristine');
 });
 
+[true, false].forEach((openOnFieldClick) => {
+    [true, false, undefined].forEach((hideOnOutsideClick) => {
+        QUnit.test(`Opened DropDownList must hide on input label click, openOnFieldClick: ${openOnFieldClick}, hideOnOutsideClick: ${hideOnOutsideClick} (T1257945)`, function(assert) {
+            const dropDownOptions = hideOnOutsideClick === undefined ? {} : { hideOnOutsideClick };
+            const $form = $('#form').dxForm({
+                formData: { CustomerID: 'VINET' },
+                items: [{
+                    itemType: 'group',
+                    colCount: 2,
+                    items: [{
+                        dataField: 'CustomerID',
+                        editorType: 'dxSelectBox',
+                        editorOptions: {
+                            items: ['VINET', 'VALUE', 'VINS'],
+                            value: '',
+                            openOnFieldClick,
+                            dropDownOptions,
+                        },
+                    }],
+                }],
+            });
+
+            const $dropDownButton = $form.find(`.${DROP_DOWN_EDITOR_BUTTON_CLASS}`);
+
+            pointerMock($dropDownButton).click();
+
+            const editorInstance = $form.dxForm('instance').getEditor('CustomerID');
+
+            assert.true(editorInstance.option('opened'), 'drop down list is visible');
+
+            const $label = $form.find(`.${FIELD_ITEM_LABEL_TEXT_CLASS}`);
+
+            pointerMock($label).click();
+
+            const willHideByInput = openOnFieldClick || (hideOnOutsideClick === false);
```

Comment:
I didn't get the second part of this expression.
Only `openOnFieldClick` says will be popup closed because of "label + editor" click or not.
Seems like you write this code just for the correct assertion below.
Can we make this part more clear?

Reply:
fixed